src/components/CDPDisplay/index.js

```diff
@@ -8,25 +8,42 @@ import useStore from 'hooks/useStore';
 import { getCdp } from 'reducers/cdps';
 import { trackCdpById } from 'reducers/multicall/cdps';
 import CDPViewPresentation from './Presentation';
+import Unavailable from '../Unavailable';
+import { Routes, ZERO_ADDRESS } from '../../utils/constants';
+import { useNavigation } from 'react-navi';
 
 function CDPView({ cdpId }) {
   cdpId = parseInt(cdpId, 10);
   const { maker, account, network } = useMaker();
   const { show: showSidebar } = useSidebar();
   const [{ cdps, feeds }, dispatch] = useStore();
   const [cdpOwner, setOwner] = useState();
+  const [cdpAvailable, setCdpAvailable] = useState(true);
+  const navigation = useNavigation();
 
   useEffect(() => {
     (async () => {
+      async function redirect(account) {
+        const { search } = (await navigation.getRoute()).url;
+        navigation.navigate({
+          pathname: `/${Routes.BORROW}/owner/${account.address}`,
+          search
+        });
+      }
       const proxyAddress = await maker
         .service('mcd:cdpManager')
         .getOwner(cdpId);
-      const cdpOwnerAddress = await maker
-        .service('proxy')
-        .getOwner(proxyAddress);
-      setOwner(cdpOwnerAddress);
+      try {
+        const cdpOwnerAddress = await maker
+          .service('proxy')
+          .getOwner(proxyAddress);
+        setOwner(cdpOwnerAddress);
+      } catch (err) {
+        account ? redirect(account) : setCdpAvailable(false);
+      }
+      if (proxyAddress !== ZERO_ADDRESS) setOwner(proxyAddress);
```

Comment:
i think this is a bug: it will call setOwner(proxyAddress) even if you just called setOwner(cdpOwnerAddress)